_posts/2018-11-13-type-constraints-correctness-reuse.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: Type constraints for code reuse and readability
+author: jguitana
+---
+
+The case for statically typed languages is so that we can prove our programs are right at compile time. Although our languages are not powerful enough to achieve this all the time, we certainly not always use their full capabilities. Our intuition should guide us to reduce implementation space and improve code reuse, allowing us to focus on small parts of a big problem. We can only do this if we choose the right abstraction in the form of Types, because these act as constraints on the implementation space, and if built correctly bring with them reusable and tested code.
+
+Let's take for example the `Int` and the `Nat` types. Int represents a number from `-x to x`. Nat represents a number from `0 to x`.
+
+*The crux of the question is*: Whenever we expect a natural number in some closure, is it different to pass an Int argument then it is passing a Nat argument?
```

Comment:
`different ... then` should be `different ... than`
or maybe rephrase to `is there difference to pass an Int or a Nat argument`

Reply:
Changed to the last phrase you mention, thanks!